crates/ruff_workspace/src/configuration.rs

```diff
@@ -486,7 +484,10 @@ impl Configuration {
                     .filter(|s| s.specificity() == spec)
                 {
                     for rule in selector {
-                        if spec == Specificity::All && !include_preview_rules && rule.is_preview() {
+                        if !matches!(spec, Specificity::Rule)
+                            && !include_preview_rules
+                            && rule.is_preview()
```

Comment:
Actually, here's another idea that I like even more...

Instead of matching on specific selector (like `--select E225`), what if we allow the selector if _all rules_ in that selector are preview? So e.g. `--select CPY` would work, as would `--select CPY001`.

Similarly, `--select E2` _would_ turn on the `E2` rules even though they're in preview. But `--select E` would not. The idea being that you'd have to explicitly select a preview category here.

Reply:
I think that @MichaReiser had a really good point about preview mode possibly changing semantic model behavior and that allowing opt-in to specific preview rules without the `--preview` flag should probably not be allowed. I think if we really want opt-in to specific preview rules, we can provide `--ignore PREVIEW` to avoid turning them all on and `--select RULE` can be used to opt-in one at a time, but the `--preview` flag would still be required and the behavior of other things could change as a consequence.

Reply:
I'm a bit undecided and realized that part of the struggle comes from that it isn't entirely clear what use cases we want to enable with preview mode. Understanding that would be important in my view because a) what I pointed out isn't a problem, b) requiring `--preview` is okay, or c) a nursery category is the better solution.

Reply:
I think it is safer to require `--preview` for any preview behavior. Once we start allowing you to opt into it without that setting, we create a lot of complexity. We can always _expand_ the behavior later if it seems like it'd be okay, but as a starting point I'd rather be restrictive.